lib/gui/components/update-notifier.js

```diff
@@ -0,0 +1,159 @@
+   * @param {String} [options.lastSleptUpdateNotifierVersion] - last slept update notifier version
+   * @param {String} options.releaseType - release type
+   * @returns {Boolean} should check for updates
+   *
+   * @example
+   * if (UpdateNotifierService.shouldCheckForUpdates({
+   *   lastSleptUpdateNotifier: Date.now(),
+   *   lastSleptUpdateNotifierVersion: '1.0.0',
+   *   releaseType: release.RELEASE_TYPE.PRODUCTION
+   * })) {
+   *   console.log('We should check for updates!');
+   * }
+   */
+  this.shouldCheckForUpdates = (options) => {
+    if (_.some([
+      !options.lastSleptUpdateNotifier,
+      options.releaseType !== release.RELEASE_TYPE.PRODUCTION,
+      packageJSON.version !== options.lastSleptUpdateNotifierVersion
+    ])) {
+      return true;
+    }
+
+    return Date.now() - options.lastSleptUpdateNotifier > units.daysToMilliseconds(this.UPDATE_NOTIFIER_SLEEP_DAYS);
+  };
+
+  /**
+   * @summary Open the update notifier widget
+   * @function
+   * @public
+   *
+   * @param {String} version - version
+   * @param {Object} [options] - options
+   * @param {Boolean} [options.allowSleepUpdateCheck=true] - allow sleeping the update check
+   * @returns {Promise}
+   *
+   * @example
+   * UpdateNotifierService.notify('1.0.0-beta.16', {
+   *   allowSleepUpdateCheck: true
+   * });
+   */
+  this.notify = (version, options = {}) => {
+    const BUTTONS = [
+      'Download',
+      'Skip'
+    ];
+
+    const BUTTON_CONFIRMATION_INDEX = _.indexOf(BUTTONS, _.first(BUTTONS));
+    const BUTTON_REJECTION_INDEX = _.indexOf(BUTTONS, _.last(BUTTONS));
+
+    const dialogOptions = {
+      type: 'info',
+      buttons: BUTTONS,
+      defaultId: BUTTON_CONFIRMATION_INDEX,
+      cancelId: BUTTON_REJECTION_INDEX,
+      title: 'New Update Available!',
+      message: `Etcher ${version} is available for download`
+    };
+
+    if (_.get(options, [ 'allowSleepUpdateCheck' ], true)) {
+      _.merge(dialogOptions, {
+        checkboxLabel: `Remind me again in ${this.UPDATE_NOTIFIER_SLEEP_DAYS} days`,
+        checkboxChecked: false
+      });
+    }
+
+    return new Bluebird((resolve) => {
+      electron.remote.dialog.showMessageBox(currentWindow, dialogOptions, (response, checkboxChecked) => {
+        return resolve({
+          agreed: response === BUTTON_CONFIRMATION_INDEX,
+          sleepUpdateCheck: checkboxChecked || false
+        });
+      });
+    }).then((results) => {
+
+      // Only update the last slept update timestamp if the
+      // user ticked the "Remind me again in ..." checkbox,
+      // but didn't agree.
+      if (results.sleepUpdateCheck && !results.agreed) {
+        settings.set('lastSleptUpdateNotifier', Date.now());
+        settings.set('lastSleptUpdateNotifierVersion', packageJSON.version);
+      }
+
+      AnalyticsService.logEvent('Close update modal', {
+        sleepUpdateCheck: results.sleepUpdateCheck,
+        notifyVersion: version,
+        agreed: results.agreed
+      });
```

Comment:
Hmmm, I wonder if it'd be worth this logging both the `notifyVersion` _and_ the `currentVersion` ? ;-)